pkg/logs/metrics/metrics.go

```diff
@@ -7,26 +7,52 @@ package metrics
 
```

Comment:
This can be a bit heavy but what about inserting interfaces in the different stages of the logs-agent, and each interface increments the two underlying metrics ? So that we keep the pipeline clean and keep the implementation details in the `logs/metrics` package ?

Reply:
That is a very interesting idea: the interface could actually be the new `telemetry.Gauge` and `telemetry.Counter` types, but we may be able to have a special implementation incrementing both expvars+the prometheus object... Do you think it would be fine if I take care of this in another PR? I want to get the telemetry feature merged for 7.17.0 and it looks to me that it could be a separated improvement PR.

Reply:
Works for me 👍